d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -207,7 +259,16 @@ public void onChanged(XdsClient.D2NodeUpdate update)
           {
             ClusterProperties clusterProperties = toClusterProperties(update.getNodeData().getData(),
                 update.getNodeData().getStat().getMzxid());
-            _clusterEventBus.publishInitialize(clusterName, clusterProperties);
+            // For symlink clusters, ClusterLoadBalancerSubscriber subscribed to the symlinks, instead of the actual node, in event bus,
```

Comment:
Curious, how does it know when to make a request for the $ prefixed symlink rather than a regular cluster?

Reply:
the service properties on ZK has a field "clusterName" which is "$xxx" for symlink-enabled cluster.